### PR TITLE
add doc for Headers.getSetCookie()

### LINKED ISSUE #24266: Document Fetch's Headers's getSetCookie() method
It was added to Fetch in https://github.com/whatwg/fetch/pull/1346.

### PR DESCRIPTION
**Pull request by chrisdavidmills**
### Description

The [`Headers.getSetCookie()`](https://fetch.spec.whatwg.org/#dom-headers-getsetcookie) method has been added to Chrome 113. This PR adds a reference document for this new method.

This is a pretty straightforward addition. You can find other useful links in my [research document](https://docs.google.com/document/d/1x1fgf7NSsrSnBw2oynXoUndK1wXehzygcegySNLs604/edit#).

One question I had — the [`Set-Cookie`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie) header page includes a note:

> **Warning:** Browsers block frontend JavaScript code from accessing the `Set-Cookie` header, as required by the Fetch spec, which defines `Set-Cookie` as a [forbidden response-header name](https://fetch.spec.whatwg.org/#forbidden-response-header-name) that [must be filtered out](https://fetch.spec.whatwg.org/#ref-for-forbidden-response-header-name%E2%91%A0) from any response exposed to frontend code.

This sounds contradictory to the purpose of `getSetCookie()` — has this note interpreted the spec correctly, and if so, should I update it to list `getSetCookie()` as an exception to this rule?

### Related issues and pull requests

Fix https://github.com/mdn/content/issues/24266

### CODE REVIEW

**Comment by Josh-Cena** [comment]
> This sounds contradictory to the purpose of `getSetCookie()` — has this note interpreted the spec correctly, and if so, should I update it to list `getSetCookie()` as an exception to this rule?

This method is intended for server environments, such as Cloudflare workers and Node.js. Client code cannot _create_ or _obtain_ a response with `Set-Cookie`, but that's not fundamentally contradictory to this method's presence on an existing response. See: https://github.com/whatwg/fetch/issues/973

**Comment by ricea** [comment]
Because the `Set-Cookie` header is stripped from responses on the network, the example as it stands is misleading. I haven't been able to come up with a compelling example, but you can do something like:
```js
const headers = new Headers({
    "Set-Cookie": "name=value",
});
headers.getSetCookie();
```

**Comment by chrisdavidmills** [comment]
> > This sounds contradictory to the purpose of `getSetCookie()` — has this note interpreted the spec correctly, and if so, should I update it to list `getSetCookie()` as an exception to this rule?
> 
> AFAICT, this method is intended for server environments, such as Cloudflare workers and Node.js. Client code cannot _create_ or _obtain_ a response with `Set-Cookie`, but that's not fundamentally contradictory to this method's presence on an existing response. See: [whatwg/fetch#973](https://github.com/whatwg/fetch/issues/973)

Thanks @Josh-Cena , that makes total sense.

**Comment by chrisdavidmills** [comment]
> Because the Set-Cookie header is stripped from responses on the network, the example as it stands is misleading. I haven't been able to come up with a compelling example, but you can do something like:

Thanks @ricea. I've updated the page now; let me know what you think

File: files/en-us/web/api/headers/getsetcookie/index.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+As alluded to above, running code like the following on the client won't return any results — `Set-Cookie` is filtered out from client-side {{domxref("Headers")}}.
+
+```js
+fetch("https://example.com").then((response) => {
+  console.log(response.headers.getSetCookie());
+  // No header values returned
+});
+```
+
+However, something like the following can be used to query `Set-Cookie` values on the server:
```
Comment (ricea) [inline files/en-us/web/api/headers/getsetcookie/index.md:42]: This will work in the browser too, so you can remove the words "on the server".

It's only when communicating with the network that the `Set-Cookie` header is removed. As long as you stay in JavaScript, you can use it (it's just not very useful).

**Review by ricea — COMMENTED** [review]
lgtm with nitpick

File: files/en-us/web/api/headers/getsetcookie/index.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+As alluded to above, running code like the following on the client won't return any results — `Set-Cookie` is filtered out from {{domxref("Headers")}} retrieved over the network.
+
+```js
+fetch("https://example.com").then((response) => {
+  console.log(response.headers.getSetCookie());
+  // No header values returned
+});
+```
+
+However, something like the following can be used to query `Set-Cookie` values. This is much more useful on the server, but it would also work on the client.
```
Comment (Josh-Cena) [inline files/en-us/web/api/headers/getsetcookie/index.md:42]: Another random drive-by: the motivation for the method is to retrieve multiple `Set-Cookie` values. Would be cool to have an example like that.

Comment (chrisdavidmills) [reply files/en-us/web/api/headers/getsetcookie/index.md]: > Another random drive-by: the motivation for the method is to retrieve multiple `Set-Cookie` values. Would be cool to have an example like that.

Makes sense; I've updated the second example to show this.

File: files/en-us/web/api/headers/getsetcookie/index.md [hunk 3]
```diff
@@ -39,15 +39,16 @@ fetch("https://example.com").then((response) => {
...
+However, the following could be used to query `Set-Cookie` values. This is much more useful on the server, but it would also work on the client.
 
 ```js
 const headers = new Headers({
-  "Set-Cookie": "name=value",
+  "Set-Cookie": "name1=value1",
+  "Set-Cookie": "name2=value2",
 });
 
 headers.getSetCookie();
```
Comment (Josh-Cena) [inline files/en-us/web/api/headers/getsetcookie/index.md]: This is against how JS object literals work: later properties always override the former, so the `Headers` constructor cannot possibly see both properties. You need to set the header later.

```js
const headers = new Headers({
  "Set-Cookie": "name1=value1",
});
headers.append("Set-Cookie", "name2=value2");
headers.getSetCookie();
```

(Can't test because my Chrome Canary doesn't seem to have this yet)

**Review by ricea — COMMENTED** [review]
lgtm, thanks!

**Comment by chrisdavidmills** [comment]
> lgtm, thanks!

@ricea super, thanks for the review.

@Elchi3 , I know you merged the BCD for this one. Can you give the method page a quick look and merge? Thanks in advance.

**Review by Elchi3 — APPROVED** [review]
Thanks Chris and everyone! 👍
